ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogClient.kt

```diff
@@ -210,31 +212,25 @@ class MogClient(val collection: MogCollection) {
     inputStream: InputStream
   ): MogMetadata {
     val startTimeMillis = System.currentTimeMillis()
-    // Read the MOG headers (not the whole file).
-    val reader = MogMetadataReader(inputStream)
-    val ifds = reader.readImageFileDirectories()
-    val imageMetadata = mutableListOf<MogImageMetadata>()
-    // IFDs are in decreasing detail (decreasing zoom), starting with max, ending with min zoom.
-    val maxZ = mogBounds.zoom + ifds.size - 1
-    ifds.forEachIndexed { i, entry ->
-      imageMetadata.add(
-        MogImageMetadata(
-          entry[TiffTag.TileWidth] as Int,
-          entry[TiffTag.TileLength] as Int,
-          mogBounds.originAtZoom(maxZ - i),
-          // TODO: Refactor casts into typed accessors.
-          entry[TiffTag.TileOffsets] as List<Long>,
-          entry[TiffTag.TileByteCounts] as List<Long>,
-          entry[TiffTag.ImageWidth] as Int,
-          entry[TiffTag.ImageLength] as Int,
-          (entry[TiffTag.JPEGTables] as? List<*>)?.map { (it as Int).toByte() }?.toByteArray()
-            ?: byteArrayOf()
+    inputStream.use { inputStreamAutoCloseable ->
+      // Read the MOG headers (not the whole file).
+      val reader = MogMetadataReader(SeekableInputStream(inputStreamAutoCloseable))
```

Comment:
Thanks for the refactor below! Can we also refactor this lambda? Perhaps we just rename this method `readMogMetadataAndClose` and call the refactored lambda `readMogMetadata`?